Add documentation retrieval for module's built-in attributes

Goose97 opened the pull request:
Closes #227

lukaszsamson commented:
That's a good first step @Goose97 but 2 more things should be covered here.
1. we need docs on attribute completion as well
2. I was thinking of extending and using BuiltinAttributes module here. It has some more documented elixir attributes that are not returned by `Module.reserved_attributes`. We need to copy docs for those from elixir repo

Goose97 commented:
I've adjusted the PR following your suggestions.

I removed `struct` from the buil-in list, because:
1. It was renamed to `__struct__`
2. This attribute is an implementation detail, so would it make more sense it we don't expose it?

lukaszsamson commented:
Thanks
